adage/analyze/templates/search/indexes/analyze/sample_text.txt

```diff
@@ -1,4 +1,6 @@
 {{ object.name }}
 {{ object.ml_data_source }}
+Experiments: {% for e in object.experiments.all %}{{ e.accession }}{% endfor %}
+Annotations:
```

Comment:
Hmmm - I'm not exactly sure how this `Experiments: ` and `Annotations: ` labels will work inside this text template, since I haven't been able to find any docs on it. Would you happen to know the location of some docs that cover this?

Reply:
I don't know of any documentation on this. The way we use these derives from the example at:

> http://django-haystack.readthedocs.io/en/master/tutorial.html#creating-searchindexes

The template comes into play in two ways:

1. the rendered template for each `Sample` is what is used to create the index with `use_template=True`, which is what we actually search via the api in `SearchResource.get_object_list` at line 69 (the part that links the template to this search is the use of the pseudo field `content`):

  ```python
  sqs = sqs.filter(content=AutoQuery(query_str)).load_all().highlight()
  ```

2. When we display the search results, the highlighted snippet that is shown is built from the text in this template.

Those two "headings" you noticed are not particularly helpful for (1), but I'm thinking they will be useful context for (2), helping the user to understand the connection between the search hit and the sample found.

I just realized that the demo site is not updated. I will do that so you can see what this looks like.

- [ ] update demo site with latest changeset

Reply:
Okay, demo site at http://greenelab.fgtech.com/ now has the latest code and has been re-indexed to show what's new.

Reply:
Right, ok, that makes sense @mhuyck